### DISCUSSION TITLE
Support i18n

### DISCUSSION THREAD

**Discussion by codingwithmanny** [wevm/vocs#91] [Ideas]
**UPDATE Feb 11/2024** - New PR being worked on #95 

# Context

Would like to see that the project supports:
1. Automatic language detection
3. Routing for different languages (assuming just folder structure /en /fr) but also allows a default without the need for a language at the end of the url (ex: /en)
4. Allows for language dropdown and easy switching and mapping between pages (ex: /en/getting-started -> /fr/getting-started)

# Solution
See example https://vitepress.dev/reference/default-theme-config#i18nrouting

**Comment by jxom** [comment]
Why bother maintaining translations when you can use this bad boy?

<img width="339" alt="Screenshot 2024-02-09 at 8 49 02 am" src="https://github.com/wevm/vocs/assets/7336481/9707f198-e75d-4a09-b6fd-5c6bde69f3f0">

**Reply by samuelstroschein** [reply]
because this bad boy doesn't give you SEO and leads to an inferior UX

**Comment by codingwithmanny** [comment]
As much as I love and dislike this tool, there's a few reasons to support multiple languages natively:

1. Google Translate isn't the best - As someone who speaks 3 languages, I can attest to this.
2. Catered educational content to a specific language
3. Not everyone is using Google Chrome to have this feature automatically enabled - also doesn't work all the time

**Reply by PSkinnerTech** [reply]
I agree with this. I speak 3 languages and two are of dialects that are not of the common core language.

Google translate is an amazing tool, but it isn't always reliable. The most difficult part is when it starts translating words that should actually stay the same since they may be used programmatically.

i18n allows for stronger docs that are written to cater to a wider audience. Obviously, more work goes into it, but the work is worth it. Especially as many products do expand deeply into Eastern countries where the Google translate results are severely lacking (i.e. China, India, Russia).

**Reply by felixhaeberle** [reply]
I second this. Once your business/project has matured, you want control over your copy/texts, and don't let a machine decide for you. Also, SEO benefits & a few more reasons. A particular bad example is Stripe. In my language (german) they have totally misleading headlines like:

<img width="606" alt="Bildschirmfoto 2024-02-09 um 14 17 26" src="https://github.com/wevm/vocs/assets/34959078/091b3fa8-184c-4e86-bade-7d7f25811681">
<img width="562" alt="Bildschirmfoto 2024-02-09 um 14 18 11" src="https://github.com/wevm/vocs/assets/34959078/d7fecdfb-fb4d-4af2-9c0d-f3ee480958ac">

The English one is talking about fast shipping (think software teams) while the German one is about shipping goods (think physical items).

**Reply by jxom** [reply]
Open to a PR!

**Reply by codingwithmanny** [reply]
New WIP PR https://github.com/wevm/vocs/pull/95

**Reply by PhABC** [reply]
How can we get this through the line? 🙏

**Reply by codingwithmanny** [reply]
@PhABC comment on the PR in support of it 🙏
